csharp/ql/test/library-tests/dataflow/flowsources/aspremote/AspRemoteFlowSource.cs

```diff
@@ -20,4 +22,33 @@ public object MyAction(ViewModel viewModel)
             throw null;
         }
     }
+
+    public class AspRoutingEndpoints
+    {
+        public delegate void MapGetHandler(string param);
+
+        public void HandlerMethod(string param) { }
+
+        public void M1(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+            var app = builder.Build();
+
+            // The delegate parameters are considered flow sources.
+            app.MapGet("/api/redirect/{newUrl}", (string newUrl) => { });
+            app.MapGet("/{myApi}/redirect/{myUrl}", (string myApi, string myUrl) => { });
+
+            Action<string> handler = (string lambdaParam) => { };
+            app.MapGet("/api/redirect/{lambdaParam}", handler);
+
+            MapGetHandler handler2 = HandlerMethod;
+            app.MapGet("/api/redirect/{mapGetParam}", handler2);
+
+            app.MapPost("/api/redirect/{mapPostParam}", (string mapPostParam) => { });
```

Comment:
I think `MapPost` (and maybe others too) works with complex models too. Can you add a test for that? Also, properties of complex models are marked as remote flow sources in https://github.com/github/codeql/pull/9393. Would that work here too?

Reply:
To make sure I understand; By complex models do you mean user defined types with fields and properties? Are you asking whether something like in the example below will be caught (I haven't tried but I don't think so)?

```csharp
public class Item
{
    public string Id { get; set; }
}

Minimal API:
...
app.MapPost("/items", (Item item) => Sink(item.Id))
app.Run();
```

Reply:
Yep, exactly. I think this might be used frequently, because even the MS samples show these at https://docs.microsoft.com/en-us/aspnet/core/tutorials/min-web-api?view=aspnetcore-6.0&tabs=visual-studio#add-the-api-code.